mrmustard/physics/representations.py

```diff
@@ -310,3 +315,178 @@ def plot(
         ax.set_title(f"${title}$")
         plt.show(block=False)
         return fig, ax
+
+
+class Fock(Representation):
+    r"""The Fock representation of a broad class of quantum states,
+    transformations, measurements, channels, etc.
+
+    The ansatz available in this representation is ArrayAnsatz.
+
+    This class allows for operations of Fock tensors.
+
+    Examples:
+        .. code-block:: python
+            array1 = math.astensor(np.random.random((1,5,7,8))) # where 1 is the batch.
+            array2 = math.astensor(np.random.random((1,5,7,8))) # where 1 is the batch.
+            array3 = math.astensor(np.random.random((3,5,7,8))) # where 3 is the batch.
+            fock1 = Fock(array1)
+            fock2 = Fock(array2)
+            fock3 = Fock(array3)
+
+            fock4 = 1.3 * fock1 - fock2 * 2.1  # linear combination can be done with the same batch dimension
+            assert fock4.ansatz.array.shape == (1,5,7,8)
+
+            fock5 = fock1 / 1.3  # support divide on a scalar
+            assert fock5.ansatz.array.shape == (1,5,7,8)
+
+            fock6 = fock1[2] @ fock3[2]  # contract wires 2 on each (inner product)
+            assert fock6.array.shape == (3, 5, 7, 5, 7) # note that the batch is 1 * 3
+
+            fock7 = fock1 & fock3  # outer product (tensor product)
+            assert fock7.array.shape == (3, 5, 7, 8, 5, 7, 8) # the batch is 1*3
+
+            fock8 = fock1.conj() # conjugate of it
+            assert fock8.array.shape == (1,5,7,8)
+    Args:
+        array (Batch[Tensor]): the array in Fock representation.
+        batch_flag (bool): if the array input has a batch dimension. False by default.
+
+    """
+
+    def __init__(self, array: Batch[Tensor], batch_flag=False):
+        self._contract_idxs: tuple[int, ...] = ()
+        if batch_flag is False:
+            array = array[None, ...]
+        self.ansatz = ArrayAnsatz(array=array)
+        self.batch_flag = batch_flag
+
+    @classmethod
+    def from_ansatz(cls, ansatz: ArrayAnsatz, batch_flag=False) -> Fock:
+        r"""Returns a Fock object from an ansatz object."""
+        return cls(ansatz.array, batch_flag=batch_flag)
+
+    @property
+    def array(self) -> Batch[Tensor]:
+        r"""
+        The array from the ansatz.
+        """
+        return self.ansatz.array
+
+    def conj(self):
+        r"""
+        The conjugate of this Fock object.
+        """
+        new = self.__class__(np.conj(self.array))
```

Comment:
```suggestion
        new = self.__class__(np.conj(self.array), batched=True)
```

Reply:
if you add a `conj` method to the `ArrayAnsatz`, you can do `new = self.from_ansatz(self.ansatz.conj)` here.